ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -1123,21 +1122,18 @@ private ServerErrorCode validateRequest(PartitionId partition, RequestOrResponse
         return ServerErrorCode.Disk_Unavailable;
       }
       // 3. check if partition exists on this node and that the store for this partition is available
-      if (storageManager.getStore(partition) == null) {
-        if (localReplica != null) {
-          // check stores on the disk
-          if (!storageManager.isDiskAvailable(localReplica.getDiskId())) {
-            metrics.diskUnavailableError.inc();
-            localReplica.markDiskDown();
-            return ServerErrorCode.Disk_Unavailable;
-          } else {
-            metrics.replicaUnavailableError.inc();
-            return ServerErrorCode.Replica_Unavailable;
-          }
-        } else {
+      ServerErrorCode errorCode = storeManager.isPartitionAvailable(partition, localReplica);
```

Comment:
If you need the error code, call the method something like checkLocalPartitionStatus.

Reply:
fixed.